[quant][refactor] Reflection pad now uses AT_DISPATCH_FLOATING_AND_QINT_TYPES

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#48051 [quant][refactor] Reflection pad now uses AT_DISPATCH_FLOATING_AND_QINT_TYPES**
* #48050 [quant] AT_DISPATCH_FLOATING_AND_QINT_TYPES
* #48037 [quant] out-variant for the reflection pad
* #48036 [quant] ReflectionPad2d

Test Plan:

Run `python test/test_quantization.py TestPadding`

Differential Revision: [D25004158](https://our.internmc.facebook.com/intern/diff/D25004158)

Commit:
[quant] Reflection pad now uses AT_DISPATCH_FLOATING_AND_QINT_TYPES

[ghstack-poisoned]

vkuzo reviewed (approved):
can we add a test plan to the diff summary

Commit:
Update on "[quant][refactor] Reflection pad now uses AT_DISPATCH_FLOATING_AND_QINT_TYPES"


Test Plan:

Run `python test/test_quantization.py TestPadding`

Differential Revision: [D25004158](https://our.internmc.facebook.com/intern/diff/D25004158)

[ghstack-poisoned]
